database_table.sql

```diff
@@ -0,0 +1,112 @@
+CREATE TABLE organizations (
+  id SERIAL PRIMARY KEY,
+  name VARCHAR(255) NOT NULL,
+  created_at TIMESTAMP NOT NULL DEFAULT NOW(),
+  updated_at TIMESTAMP NOT NULL DEFAULT NOW()
+);
+
+-- Members table
+CREATE TABLE members (
+  id SERIAL PRIMARY KEY,
+  username VARCHAR(255) NOT NULL,
+  email VARCHAR(255) NOT NULL,
+  password VARCHAR(255) NOT NULL,
+  created_at TIMESTAMP NOT NULL DEFAULT NOW(),
+  updated_at TIMESTAMP NOT NULL DEFAULT NOW()
+);
+
+-- Organization members table
+CREATE TYPE roles as ENUM('organization_head', 'member');
+CREATE TABLE organization_members (
+  organization_id INT REFERENCES organizations(id),
+  member_id INT REFERENCES members(id),
+  organization_member_role roles ,
+  created_at TIMESTAMP NOT NULL DEFAULT NOW(),
+  updated_at TIMESTAMP NOT NULL DEFAULT NOW()
+);
+-- Plans table
+CREATE TABLE plans (
+  id SERIAL PRIMARY KEY,
+  name VARCHAR(255) NOT NULL,
+  duration INT NOT NULL,
+  created_at TIMESTAMP NOT NULL DEFAULT NOW(),
+  updated_at TIMESTAMP NOT NULL DEFAULT NOW()
+);
+
+-- Organization plans table
+CREATE TABLE organization_plans (
+  organization_id INT REFERENCES organizations(id),
+  plan_id INT REFERENCES plans(id),
+  created_at TIMESTAMP NOT NULL DEFAULT NOW(),
+  updated_at TIMESTAMP NOT NULL DEFAULT NOW()
+);
+
+-- Channels table
+CREATE TYPE channel_type as ENUM('Personal', 'Group');
+CREATE TABLE channels (
+  id SERIAL PRIMARY KEY,
+  channel_type channel_type;
+  name VARCHAR(255) NOT NULL,
+  created_at TIMESTAMP NOT NULL DEFAULT NOW(),
+  updated_at TIMESTAMP NOT NULL DEFAULT NOW()
+);
+-- Organization_Channel
+CREATE TABLE organization_channel(
```

Comment:
Not sure if you need join table to associate channel and organisation. Can channel belong to multiple organisations?

Reply:
I used Discord's data schema as a reference. They do not have separate tables for group or personal chats. Instead, they use the 'channel_type' attribute to determine whether a chat is personal or group. Therefore, I created the 'organization_channel' table to get a list of all the channels belonging to a particular organization.